web/src/emoji.ts

```diff
@@ -268,27 +337,34 @@ export function get_emoji_details_for_rendering(opts) {
     };
 }
 
-function build_default_emoji_aliases({names, get_emoji_codepoint}) {
+function build_default_emoji_aliases({
+    names,
+    get_emoji_codepoint,
+}: {
+    names: string[];
+    get_emoji_codepoint: (name: string) => string | undefined;
+}): Map<string, string[]> {
     // Please keep this as a pure function so that we can
     // eventually share this code with the mobile codebase.
 
     // Create a map of codepoint -> names
-    const map = new Map();
+    const map = new Map<string, string[]>();
 
     for (const name of names) {
         const base_name = get_emoji_codepoint(name);
-
-        if (map.has(base_name)) {
-            map.get(base_name).push(name);
-        } else {
-            map.set(base_name, [name]);
+        if (base_name !== undefined) {
```

Comment:
It's definitely a server/client interface bug if this isn't true, so we should write this as:
```
if (base_name === undefined) {
    blueslip.error(`No codepoint for emoji name ${name}`);
    continue;
}
```

and leave the existing code as-is.

Reply:
Will it make sense to do a null assertion here? Like so `const base_name = get_emoji_codepoint(name)!;` because I think `build_default_emoji_aliases` call gets called at initialization only so it's free from any user input and so I guess in this function `get_emoji_codepoint` will always return a valid codepoint?

Reply:
Ideally we will want to guard against invalid data from the server too. We aren’t consistent about that yet, but we will need to become consistent about it in order to let the frontend run securely in a future standalone mode. So we should avoid baking in assumptions that the server always sends valid data.

Reply:
Also, even without that consideration, the chain of reasoning one needs to go through in order to show that this isn’t undefined is quite long, and could easily be broken by bugs far away from this code. We should generally reserve non-null assertions for cases where that reasoning is short and local.

Reply:
Okay makes sense, I will do this change then. But I will have to probably change some tests as well for maintaining 100% test coverage.